lib/features/keyboard-move-selection/ChoreoMoveSelection.js

```diff
@@ -18,14 +18,22 @@ export default function ChoreoMoveSelection(injector, selection) {
       return;
     }
 
-    // participant bands and messages are not movable
+    // Participant bands and messages are not movable unless their activities are also selected
     let isNotMovable = function(shape) {
-      return is(shape, 'bpmn:Participant') || is(shape, 'bpmn:Message');
+      const participantActivityUnselected = is(shape, 'bpmn:Participant') && !selectedElements.includes(shape.parent);
+      const messageActivityUnselected = is(shape, 'bpmn:Message') && !selectedElements.includes(shape.parent.parent);
+
+      return participantActivityUnselected || messageActivityUnselected;
     };
+
     if (selectedElements.some(isNotMovable)) {
       return;
     }
 
+    // We remove all these shapes as they will be moved anyway due to being descendant of the activity
+    selection._selectedElements = selectedElements.filter(
```

Comment:
Can't we do a `set()` or something on `selection`? This seems hacky when compared to the `get()` above.

Reply:
Unfortunately, there is no ```set()``` on ```selection```.